Separated the NetModule to dependant and standalone

pokk opened the pull request:
as the title.

pokk committed:
Separated the `NetModule` to dependant and standalone.

pokk committed:
Separated the stand alone and the dependent model.

pokk committed:
Separated the di module and component.

pokk committed:
Merge remote-tracking branch 'origin/develop/master' into develop/dependent_standalone

pokk committed:
Merged the newest commit and resolved the conflict.

pokk commented:
@lekaha 

Please take a look. 🙏

pokk committed:
Reverted the NetModule, PlayoneModule, and ApplicationComponent.

pokk committed:
Fixed the typo variable.

lekaha commented on mobile-ui/src/standalone/java/com/playone/mobile/ui/firebase/StandAloneAuthenticator.kt:
```diff
@@ -0,0 +1,34 @@
+import com.playone.mobile.domain.model.User
+
+class StandAloneAuthenticator : Authenticator() {
+    val mapper = FirebaseUserMapper()
```
empty line please

lekaha commented on mobile-ui/src/standalone/java/com/playone/mobile/ui/firebase/StandAloneAuthenticator.kt:
```diff
@@ -0,0 +1,34 @@
+import com.playone.mobile.domain.Credential
+import com.playone.mobile.domain.model.User
+
+class StandAloneAuthenticator : Authenticator() {
```
`Standalone` is a word so I think it is better do not use `StandAlone`

lekaha commented on mobile-ui/src/standalone/java/com/playone/mobile/ui/firebase/StandAloneAuthenticator.kt:
```diff
@@ -0,0 +1,34 @@
+    class FirebaseUserMapper {
+
+        fun mapToUser(user: FirebaseUser) =
+            User().apply {
```
empty line!

lekaha commented on mobile-ui/src/standalone/java/com/playone/mobile/ui/firebase/StandAloneAuthenticator.kt:
```diff
@@ -0,0 +1,34 @@
+
+    override fun isVerifiedEmail() = TODO()
+
+    class FirebaseUserMapper {
```
I think this Mapper is not needed for Standalone

pokk committed:
Fixed the issues.

1. https://github.com/lekaha/playone-android/pull/56#discussion_r173061803
2. https://github.com/lekaha/playone-android/pull/56#discussion_r173061854
3. https://github.com/lekaha/playone-android/pull/56#discussion_r173063987

pokk commented:
@lekaha 

I've fixed! Please take a look again. 🙏
